src/providers/xref.provider.ts

```diff
@@ -30,15 +30,41 @@ function shouldProvide (context: Context): boolean {
 }
 
 async function getLabels (): Promise<string[]> {
-  const regex = /\\[\\[(\\w+)\\]\\]/g
-  const labels = await vscode.workspace.findFiles('**/*.adoc').then((files) =>
-    files
-      .map((uri) => readFileSync(uri.path).toString('utf-8'))
-      .join('\n')
-      .match(regex)
-      .map((result) => result.replace('[[', '').replace(']]', ''))
-  )
-  return labels
+  const files = await vscode.workspace.findFiles('**/*.adoc')
+  const contentOfFilesConcatenated = files
+    .map((uri) => readFileSync(uri.path).toString('utf-8'))
+    .join('\n')
+  const labelsFromLegacyBlock = await getLabelsFromLegacyBlock(contentOfFilesConcatenated)
+  const labelsFromShorthandNotation = await getLabelsFromShorthandNotation(contentOfFilesConcatenated)
+  const labelsFromLonghandNotation = await getLabelsFromLonghandNotation(contentOfFilesConcatenated)
+  return labelsFromLegacyBlock.concat(labelsFromShorthandNotation, labelsFromLonghandNotation)
+}
+
+async function getLabelsFromLonghandNotation (content: string): Promise<string[]> {
+  const regex = /\[id=(\w+)\]/g
```

Comment:
I think we need to check that the line starts with `[` using `^`

Reply:
Seems there is a case where it is not at the begiining of the line, for inlined anchor:
https://docs.asciidoctor.org/asciidoc/latest/attributes/id/#as-an-inline-anchor

```
* First item
* [[step2]]Second item
* Third item
```

Reply:
also for additional anchors to a section https://docs.asciidoctor.org/asciidoc/latest/attributes/id/#add-additional-anchors-to-a-section

```
[#version-4_9]
=== [[current]][[latest]]Version 4.9
```

Reply:
You are right... seems like we need a better parsing strategy if we want to be 100% accurate.